ROADMAP.md

```diff
@@ -1,25 +1,389 @@
+* Bundle format includes attestation support that interops with SLSA
+* Reference implementation of how to use signature verification across various points in the software development lifecycle, leveraging policy-controller as one example
+
+## Signature Transparency Log (Rekor)
+
+### Short Term (< 6 mo)
+
+* Add 2 independent log monitors
+    * Monitors will only verify log consistency
+* Witnessing
+    * Monitors should transition into fully fledged witnesses
+    * Witnesses should publish signed checkpoints to distributor
+* V1 for GitHub Actions-based identity monitor
+* Timestamp at Entry Upload
+    * Timestamp should be provided by TSA
+    * Timestamp becomes immutable and verifiable
+* Complete architecture document
+* Publish inclusion events to pub/sub queueo
+
+### Medium Term (6-12 mo)
+
+* API v2 Design & Implementation
+    * Reduce API surface to be structured around key types, push parsing logic to clients
+    * Must support witnessing without compromises
+* MVP split-view attack mitigation
+    * Client requests witnessed checkpoints from a set of witnesses or distributors
+    * Client verifies inclusion proof from log using checkpoint size
+    * Offline verifiable proofs include witnessed checkpoints in Bundle
+* Verifiable Search Design
+    * Detecting credential compromise without requiring constant log scraping
+* Design transparent redaction mechanism
+* Reviewing sharding strategy to better align with Certificate Transparency's sharding
+* Experiment with serverless/log tile format for caching and scalability
+* Design log replication mechanism for scaling instances (possibly can use serverless)
+
+### Long Term (12+ mo)
+
+* Verifiable Search Implementation
+* Package repositories participate in mitigating split-views
+    * Witnessed checkpoints attached to artifact signature metadata
+    * Repositories verify log consistency
+* Implementation for log replication
+* Transparent redaction implementation
+* Privacy-preserving log queries, where the log is not aware of client's artifact of interest
+* Roughtime for distributed trusted timestamps
+* Continued fuzzing of large API surface
+
+## Certificate Authority (Fulcio)
+
+### Short Term (< 6 mo)
+
+* Hashed email addresses/PII
+    * Still supports monitoring identities
+* Complete architecture document
+* Fulcio CI identity provider onboarding automation
+* Standardize OIDC identity provider requirements for inclusion in public instance
+
+### Medium Term (6-12 mo)
+
+* Privacy-preserving identities
+    * VRFs or ZKP to prevent attacker from learning when a signing event occurred for a given identity
+    * Must still support identity monitoring
+* Zero-knowledge proofs that prove a CA verified an identity token
+
+### Long Term (12+ mo)
+
+* ACME for code-signing certificates (ongoing RFC)
+* Native SPIFFE support with X509-SVIDs as authenticators, rather than supporting only OIDC
+* Support revocation for private deployments
```

Comment:
is revocation for the public deployment intentionally omitted? This could be useful for key or account compromise (discussed in #sigstore/cosign#677)

Reply:
It's a hard problem on a large scale - How do we authenticate requests to revoke content? How do we maintain an indefinitely growing list? WebPKI doesn't deal with revocation well, and I worry we'd run into the same issues.

Reply:
I agree that this is challenging, and should be part of the 12 months+ roadmap. But if Rekor is used to discover a mis-behaving key/identity, Sigstore should do *something* about this. Whether this is a revocation list, TUF delegations to trusted identities, or some other mechanism.

Reply:
Is this revoking specific entries in the log by UUID, revoking everything signed with public key X, revoking everything signed by x509 cert with serial Y, or revoking everything signed by `foo@bar.com` between a date range?

Reply:
I that case I agree. I think it is up to the consumers to determine what is trustworthy. 

I imagine package manager services would manage this in most cases so it would be transparent to the user, and the operators of each package service can handle authz for determining who can revoke trust.

Reply:
I agree that the policy should be up to consumers, but Sigstore can provide tooling to make these revocations possible

Reply:
@mnm678 Could the PKI for public deployments (including policies) be covered by [Line 380](https://github.com/sigstore/community/pull/315/files#diff-683343bdf93f55ed3cada86151abb8051282e1936e58d4e0a04beca95dff6e51R380)?

Reply:
possibly, depending on how that is implemented

Reply:
Do you have suggestions of wording changes to either this bullet point or the one Trishank noted that would cover what you're looking for? A mention of "client support for revocation policies"?

Reply:
maybe "tooling to support client revocation policies"

Reply:
I'd say it's more general than just revocation, but something along the lines of: "A solution for distributing, revoking, and rotating public keys and policies used to verify public artifacts"

Reply:
I think that fits under the client roadmap with "Policy: sophisticated verification policies, or deeper integration with a policy evaluation service".

Reply:
I suggest we don't use the word "revocation" until we had a discussion about it. Avoiding any "revocation" solution is generally a good thing, since most of them has not worked very well. So Hayden's suggestion seems good.

Reply:
Right, that makes sense: each, say, package registry may wish to handle revocation separately from Sigstore. Sigstore should ideally "just" be a (but not the only) source of truth.

Reply:
Do we have a resolution about this? @mnm678 @haydentherapper

Reply:
Hayden's interpretation of it fitting under the policy item works for me